Control/public/Model.js

```diff
@@ -297,14 +297,23 @@ export default class Model extends Observable {
   }
 
   /**
-   * After pages load check if the user enabled notifications
+   * Request notification permission
    */
-  checkBrowserNotificationPermissions() {
+  requestBrowserNotificationPermissions() {
     if (Notification.permission !== 'granted') {
       Notification.requestPermission();
     }
   }
 
+  /**
+   * Check whether it makes sense to ask for notification permission
+   */
+  checkBrowserNotificationPermissions() {
+    return (Notification.permission === 'denied' ||
+      Notification.permission === 'default') &&
+      window.location.protocol == "https:";
+  }
```

Comment:
2 small details: if it is not too much work maybe you can put a different name, here I would expect it to return true if the notification permissions **is** granted
Also for consistency the third comparison may use strict comparison